app/main.py

```diff
@@ -1,54 +1,73 @@
-import json
-import xml.etree.ElementTree as ET
+from app.display import ConsoleDisplay, ReverseDisplay, Display
+from app.print import ConsolePrinter, ReversePrinter, Printer
+from app.serializers import JsonSerializer, XmlSerializer, Serializer
 
 
 class Book:
-    def __init__(self, title: str, content: str):
+    def __init__(
+            self,
+            title: str,
+            content: str,
+            display: Display = ConsoleDisplay(),
+            printer: Printer = ConsolePrinter(),
+            serializer: Serializer = JsonSerializer()
+    ) -> None:
         self.title = title
         self.content = content
+        self.display_service = display
+        self.printer_service = printer
+        self.serialize_service = serializer
 
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
-
-
-def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+    def display(self) -> None:
```

Comment:
Indicate that you expect to obtain a display argument of type `Display` and call a corresponding method from it. Add a similar change in the other methods. It would help to simplify the `main` function logic as well